agent/containermetadata/parse_metadata.go

```diff
@@ -0,0 +1,153 @@
+	}
+
+	// Get Port bindings from docker configurations
+	var ports []api.PortBinding
+	var err error
+	if dockerContainer.HostConfig != nil {
+		ports, err = api.PortBindingFromDockerPortBinding(dockerContainer.HostConfig.PortBindings)
+		if err != nil {
+			seelog.Warnf("Failed to parse container metadata for task %s container %s: %v", task, container, err)
+		}
+	} else {
+		seelog.Warnf("Failed to parse container metadata for task %s container %s: container has no host configuration", task, container)
+	}
+
+	networkMetadata, err := parseNetworkMetadata(dockerContainer.NetworkSettings, dockerContainer.HostConfig)
+	if err != nil {
+		seelog.Warnf("Failed to parse container metadata for task %s container %s: %v", task, container, err)
+	}
+
+	return DockerContainerMetadata{
+		containerID:         dockerContainer.ID,
+		dockerContainerName: dockerContainer.Name,
+		imageID:             dockerContainer.Image,
+		imageName:           imageNameFromConfig,
+		ports:               ports,
+		networkInfo:         networkMetadata,
+	}
+}
+
+// parseTaskMetadata parses  in the AWS configuration and task
+// and packages this data for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseTaskMetadata(task *api.Task, container *api.Container) TaskMetadata {
+	containerName := container.Name
+
+	taskARNFromConfig := task.Arn
+
+	return TaskMetadata{
+		containerName: containerName,
+		taskARN:       taskARNFromConfig,
+	}
+}
+
+// parseMetadata gathers metadata from a docker container, and task
+// configuration and data then packages it for JSON Marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this or the above stage.
+func (manager *metadataManager) parseMetadata(dockerContainer *docker.Container, task *api.Task, container *api.Container) Metadata {
+	taskMD := parseTaskMetadata(task, container)
+	dockerMD := parseDockerContainerMetadata(task, container, dockerContainer)
+	return Metadata{
+		cluster:                 manager.cluster,
+		taskMetadata:            taskMD,
+		dockerContainerMetadata: dockerMD,
+		containerInstanceARN:    manager.containerInstanceARN,
+		metadataStatus:          "READY",
+	}
+}
+
+// parseMetadataAtContainerCreate gathers metadata from task and cluster configurations
+// then packages it for JSON Marshaling. We use this version to get data
+// available prior to container creation
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this or the above stage.
+func (manager *metadataManager) parseMetadataAtContainerCreate(task *api.Task, container *api.Container) Metadata {
+	taskMD := parseTaskMetadata(task, container)
+	return Metadata{
+		cluster:              manager.cluster,
+		taskMetadata:         taskMD,
+		containerInstanceARN: manager.containerInstanceARN,
+		metadataStatus:       "NOT_READY",
```

Comment:
Could we have a new type for metadata status ?
I'd like to have it be one amongst a set of well defined values